src/main/resources/com/suse/matcher/rules/drools/InputAugmenting.drl

```diff
@@ -38,3 +38,101 @@ rule "generatePenaltyGroups"
         insert(new PenaltyGroup(penaltyGroupId, $guestId));
 end
 
+// aggregate 2 equal (same properties, same product set) OEM subscriptions into a single one
+rule "aggregateEqualStackableSubscriptions"
+    agenda-group "InputAugmenting"
+    when
+        $s1: Subscription(ignored == false, stackable == true, id < 0, aggregationSubscriptionId == null || aggregationSubscriptionId == id)
+
+        // make sure that the $s1 is the one that has lowest id
+        not Subscription(
+            ignored == false,
+            stackable == true,
+            id < 0, // OEM subscription
+            id < $s1.id,
+            partNumber == $s1.partNumber,
+            name == $s1.name,
+            policy == $s1.policy,
+            cpus == $s1.cpus,
+            sccUsername == $s1.sccUsername,
+            quantity == 1)
+
+        $s2: Subscription(
+            ignored == false,
+            stackable == true,
+            id < 0, // OEM subscription
+            id > $s1.id,
```

Comment:
IIUC a simpler way to implement this mechanism is to aggregate all subscription couples `($s1, $s2)` where `$s1.id < $s2.id`. Eventually they will converge to the minimum `$s1.id`, even without explicitly searching for it (`not Subscription` clause above).

Does that make sense?

Reply:
Yes, that was originally the case. The problem with this that any such intermediate merging can produce a set of `adjust_pinned_match` messages, which aren't valid after the intermediate aggregated subscription is merged later. Let me think about it, I also don't like having that `not ...` clause there...

Reply:
Ok, maybe we could write a separate rule for generating the `adjust_pinned_match`, but again - we'd need to prevent activating it before merging of all mergeable subscriptions is complete.

Reply:
No, without using `salience` or other agenda I don't see how this could be solved easily.

Reply:
Ok, in the end I rewrote the rules to use `salience` for the "cleanup" rules (merging pinned matches + deleting stale stuff). It looks cleaner, I'll push the changes today.